lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -559,7 +559,10 @@ def get_blob_client(self, resource_group_name, storage_account_name):
 
         try:
             self.log('Create blob service')
-            return CloudStorageAccount(storage_account_name, account_keys.keys[0].value).create_block_blob_service()
+            if storage_blob_type == 'page':
+                return CloudStorageAccount(storage_account_name, account_keys.keys[0].value).create_page_blob_service()
+            else:
```

Comment:
Should explicitly check for `blob` here and have the default `else` bomb out, so if a caller passes something nonsensical, we error instead of silently succeeding with bogus input.